src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -61,148 +48,16 @@ public FindAllReferencesHandler(IThreadingContext threadingContext, IMetadataAsS
             var findUsagesService = document.GetRequiredLanguageService<IFindUsagesLSPService>();
             var position = await document.GetPositionFromLinePositionAsync(
                 ProtocolConversions.PositionToLinePosition(referenceParams.Position), cancellationToken).ConfigureAwait(false);
-            var context = new SimpleFindUsagesContext(cancellationToken);
 
-            // Finds the references for the symbol at the specific position in the document, pushing the results to the context instance.
-            await findUsagesService.FindReferencesAsync(document, position, context).ConfigureAwait(false);
-
-            return await GetReferenceItemsAsync(document, position, context, cancellationToken).ConfigureAwait(false);
-        }
-
-        private async Task<LSP.VSReferenceItem[]> GetReferenceItemsAsync(
-            Document document,
-            int position,
-            SimpleFindUsagesContext context,
-            CancellationToken cancellationToken)
-        {
-            // Mapping each reference to its definition
-            var definitionMap = new Dictionary<DefinitionItem, List<SourceReferenceItem>>();
-            foreach (var reference in context.GetReferences())
-            {
-                if (!definitionMap.ContainsKey(reference.Definition))
-                {
-                    definitionMap.Add(reference.Definition, new List<SourceReferenceItem>());
-                }
-
-                definitionMap[reference.Definition].Add(reference);
-            }
-
-            // NOTE: Parts of FAR currently do not display correctly due to a bug in LSP.VSReferenceItem.
-            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1088938/
-            using var _ = ArrayBuilder<LSP.VSReferenceItem>.GetInstance(out var referenceItems);
-
-            // Each reference is assigned its own unique id
-            var id = 0;
-            foreach (var (definition, references) in definitionMap)
-            {
-                // Creating the reference item that corresponds to the definition.
-                var definitionItem = await GenerateReferenceItem(
-                    id, definitionId: id, document, position, definition.SourceSpans.FirstOrDefault(), context, definition.DisplayableProperties,
-                    definition.GetClassifiedText(), symbolUsageInfo: null, cancellationToken).ConfigureAwait(false);
-
-                // If we have an empty location, skip this definition and its references.
-                if (definitionItem.Location == null)
-                {
-                    continue;
-                }
-
-                referenceItems.Add(definitionItem);
-                var definitionId = id;
-                id++;
+            var context = new FindUsagesLSPContext(
+                referenceParams.PartialResultToken, document, position, _metadataAsSourceFileService, cancellationToken);
```

Comment:
do you need to close/dispose either referenceParams or .PartialResulttoken?

Reply:
I don't believe so.